portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,33 +46,46 @@ private String convertToJsonUsingGson(List<String> messages) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get comments in datastore, by most recent order at the top.
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    // Iterate over all entities, get comment text.
-    List<String> comments = new ArrayList<>();
+    // Extract limit on number of comments from query string.
+    Integer limit = Integer.parseInt(request.getParameter("limit"));
```

Comment:
I think it's probably good to think the failure situations here, especially since in the real world you might not always be writing all the requests.

request.getParameter could return a 'null' if the "limit" parameter doesn't exist, and calling .parseInt(null) will cause your program to crash. It's also possible that the request.getParameter call returns an empty string, and .parseInt("") will also throw an error.

So this method requires that there be a limit parameter set to a non-empty value. That's an important requirement that either needs to be strongly communicated to the users or you need to add some checks to make your code more robust so it doesn't crash in these scenarios.

Reply:
Ok, I created a ternary statement that first verifies whether or not that parameter exists -- if not, then a default value is assigned.